### PR TITLE
Add Reseller company

### PR DESCRIPTION
**Pull request by gwmoura**
* Add one module named Company
* Create CRUD for Manager add one Reseller Company
* Add tests to Reseller
* Add factories to Reseller

### CODE REVIEW

**Comment by fernandes** [comment]
its damm equal the Manager::Company

the database persistence is ok to keep duplicated, but please extract common behavior in classes (for exemple document) the same for rspec in shared examples :+1:

**Comment by gwmoura** [comment]
@fernandes check this update!

File: app/models/entities/company.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module Entities::Company
```
Comment (fernandes) [inline app/models/entities/company.rb]: where is test for this module?

File: app/models/entities/manager/company.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 # Person(natural person) - is a real individual
 # Company(legal person) - is a company, business, entity, association, institution
 class Entities::Manager::Company < ActiveRecord::Base
+  include Entities::Company
```
Comment (fernandes) [inline app/models/entities/manager/company.rb]: why do you include if the module add no method on the class (at least the module has only class methods and you're calling them directly here :point_down:

File: app/models/entities/manager/company.rb [hunk 3]
```diff
@@ -55,27 +52,15 @@ class Create < Trailblazer::Operation
...
-        end
-
-        errors.add(:document, "is not valid") unless document_spec.valid?(document)
-      end
-
-      private
-
-      def document_spec
-        Entities::Document.finder('br', 'Company')
+        errors.add(:document, "is not valid") unless Entities::Company.valid_document? document
```
Comment (fernandes) [inline app/models/entities/manager/company.rb]: :point_right:

File: app/models/entities/manager/company.rb [hunk 4]
```diff
@@ -88,22 +73,12 @@ class Manager < self
       # @option params [String] :document
       # @option params [String] :email
       def process(params)
-        params = sanitize_document(params)
+        params = Entities::Company.sanitize_document(params)
         validate(params[:manager_company]) do |f|
```
Comment (fernandes) [inline app/models/entities/manager/company.rb]: :point_right:

File: app/models/entities/manager/reseller.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
+# An organization can contain several members
+# where one is the owner
+# An organization is an entity / institution and can be of two types:
+# Person(natural person) - is a real individual
+# Company(legal person) - is a company, business, entity, association, institution
+class Entities::Manager::Reseller < ActiveRecord::Base
+  include Entities::Company
```
Comment (fernandes) [inline app/models/entities/manager/reseller.rb]: same comments from `Entities::Manager::Company` here

File: app/models/entities/manager/reseller.rb [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  # end
+
+  private
+  # Operation that creates the organization
+  class Create < Trailblazer::Operation
+    include Model, Responder, Common::OpBuild
+    model Entities::Manager::Reseller, :create
+
+    # Contract
+    contract do
```
Comment (fernandes) [inline app/models/entities/manager/reseller.rb]: I'm gonna merge https://github.com/railsblocks/railsblocks-entities/pull/5 then you apply the concept of isolating profile here

**Comment by gwmoura** [comment]
@fernandes check the updates

File: app/models/entities/manager/company.rb (hunk 3 above)
Comment (gwmoura) [inline app/models/entities/manager/company.rb]: In my tests, valid_document? not run in contract, the method valid_document? was called by a Reform::Form not a Create class.

The solution found is this

You have sugestions?

Comment (fernandes) [inline app/models/entities/manager/company.rb]: the contract is a `Reform::Form` object

I didn't understand whats your question

File: app/models/entities/company.rb (hunk 1 above)
Comment (fernandes) [inline app/models/entities/company.rb]: https://github.com/railsblocks/railsblocks-entities/pull/6/files#diff-9327cf697eddf17ea9d15693a0039272

good :+1:

**Comment by fernandes** [comment]
seems good to move forward, good job @gwmoura ! :smiley_cat:

**Pull request merged**